tools/lint/src/main/kotlin/NonAndroidxNullabilityDetector.kt

```diff
@@ -0,0 +1,136 @@
+import org.jetbrains.uast.UDeclaration
+import org.jetbrains.uast.UElement
+import org.jetbrains.uast.UField
+import org.jetbrains.uast.UMethod
+
+private val NULLABLE_ANNOTATIONS = listOf("Nullable", "CheckForNull")
+private val NOT_NULL_ANNOTATIONS = listOf("NonNull", "NotNull", "Nonnull")
+
+private val ANDROIDX_ANNOTATIONS = listOf(
+        "androidx.annotation.Nullable",
+        "androidx.annotation.NonNull",
+        "android.support.annotation.Nullable",
+        "android.support.annotation.NonNull")
+
+class NonAndroidxNullabilityDetector : Detector(), SourceCodeScanner {
+    companion object Issues {
+        private val IMPLEMENTATION = Implementation(
+                NonAndroidxNullabilityDetector::class.java,
+                Scope.JAVA_FILE_SCOPE
+        )
+
+        @JvmField
+        val NON_ANDROIDX_NULLABILITY = Issue.create(
+                id = "FirebaseNonAndroidxNullability",
+                briefDescription = "Use androidx nullability annotations.",
+
+                explanation = "Use androidx nullability annotations instead.",
+                category = Category.COMPLIANCE,
+                priority = 1,
+                severity = Severity.ERROR,
+                enabledByDefault = false,
+                implementation = IMPLEMENTATION
+        )
+    }
+
+    override fun getApplicableUastTypes(): List<Class<out UElement>>? {
+        return listOf(UClass::class.java, UMethod::class.java, UField::class.java)
+    }
+
+    override fun createUastHandler(context: JavaContext): UElementHandler? {
+        // using deprecated psi field here instead of sourcePsi because the IDE
+        // still uses older version of UAST
+        if (isKotlin(context.uastFile?.sourcePsi)) {
+            // These checks apply only to Java code
+            return null
+        }
+        return Visitor(context)
+    }
+
+    class Visitor(private val context: JavaContext) : UElementHandler() {
+        override fun visitClass(node: UClass) {
+            doVisit(node)
+        }
+
+        override fun visitMethod(node: UMethod) {
+            doVisit(node)
+            for (parameter in node.uastParameters) {
+                doVisit(parameter)
+            }
+        }
+
+        override fun visitField(node: UField) {
+            doVisit(node)
+        }
+
+        private fun doVisit(node: UDeclaration) {
+            for (annotation in node.annotations) {
+                if (isNullabilityAnnotation(annotation)) {
+                    ensureAndroidNullability(context, annotation)
+                }
+            }
+        }
+
+        private fun ensureAndroidNullability(context: JavaContext, annotation: UAnnotation) {
+            annotation.qualifiedName?.let { name ->
+                if (name in ANDROIDX_ANNOTATIONS) {
+                    return
+                }
+
+                val simpleName = name.split('.').last()
+                val replacement = if (simpleName in NOT_NULL_ANNOTATIONS) "NonNull" else "Nullable"
```

Comment:
Reading this I initially thought this was a bug (thinking you might replace all annotations with `Nullable`) until I realized that this method is gated by `isNullabilityAnnotation`.

That method is essentially doing everything you're doing in this method so consider just inlining here as another guard condition.

Reply:
done